sdk/metric/correct_test.go

```diff
@@ -187,6 +187,70 @@ func TestSDKLabelEncoder(t *testing.T) {
 	require.Equal(t, `[{A {8 0 B}} {C {8 0 D}}]`, labels.Encoded())
 }
 
+func TestSDKLabelsDeduplication(t *testing.T) {
+	ctx := context.Background()
+	cagg := counter.New()
+	batcher := &correctnessBatcher{
+		t:   t,
+		agg: cagg,
+	}
+	sdk := sdk.New(batcher, sdk.NewDefaultLabelEncoder())
+
+	counter := sdk.NewInt64Counter("counter")
+
+	const (
+		maxKeys = 21
+		keySets = 2
+		repeats = 3
+	)
+	var keysA []core.Key
+	var keysB []core.Key
+
+	for i := 0; i < maxKeys; i++ {
+		keysA = append(keysA, core.Key(fmt.Sprintf("A%03d", i)))
+		keysB = append(keysB, core.Key(fmt.Sprintf("B%03d", i)))
+	}
+
+	var allExpect [][]core.KeyValue
+	for numKeys := 0; numKeys < maxKeys; numKeys++ {
+
+		var kvsA []core.KeyValue
+		var kvsB []core.KeyValue
+		for r := 0; r < repeats; r++ {
+			for i := 0; i < numKeys; i++ {
+				kvsA = append(kvsA, keysA[i].Int(r))
+				kvsB = append(kvsB, keysB[i].Int(r))
+			}
+		}
+
+		var expectA []core.KeyValue
+		var expectB []core.KeyValue
+		for i := 0; i < numKeys; i++ {
+			expectA = append(expectA, keysA[i].Int(repeats-1))
+			expectB = append(expectB, keysB[i].Int(repeats-1))
+		}
+
+		counter.Add(ctx, 1, sdk.Labels(kvsA...))
```

Comment:
Repeat the previous line to prove that `sdk.Labels(kvsA...)` called multiple times produces the same LabelSet.Ordered(), or that the internal map works. So you will receive value 2 for every unique LabelSet.

Reply:
This is the new test, which I'm sure validates the change. You asked to extend the test to test the value correctness (which I have done), but I assert that the existing test is good enough. It's simply checking that the number of unique value sets is correct.